src/burn/engine/search.cpp

```diff
@@ -1237,6 +1243,7 @@ static HRESULT MsiProductSearch(
     case BURN_MSI_PRODUCT_SEARCH_TYPE_LANGUAGE:
         type = BURN_VARIANT_TYPE_STRING;
         break;
+    case BURN_MSI_PRODUCT_SEARCH_TYPE_EXISTS: __fallthrough;
```

Comment:
Rather than fall through here and check the type again below (line 1255), we should just set the value appropriately in this `case`.

Reply:
The value is in this stage string format?

Reply:
Do you want me to anything with this commit?
I am not using existing case because at this point the value is string ang later converted to int.

    hr = BVariantChangeType(&value, type);

and then int value is compared

    value.llValue = (value.llValue == INSTALLSTATE_ABSENT) ? 0 : 1;